PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -1,150 +1,186 @@
-                             : edm::EDGetTokenT<edm::ValueMap<float>>()),
-      PuppiWeightNoLep_(usePuppi_ ? consumes<edm::ValueMap<float>>(iConfig.getParameter<edm::InputTag>("PuppiNoLepSrc"))
-                                  : edm::EDGetTokenT<edm::ValueMap<float>>()),
-      PuppiCandsMap_(usePuppi_
-                         ? consumes<edm::ValueMap<reco::CandidatePtr>>(iConfig.getParameter<edm::InputTag>("PuppiSrc"))
-                         : edm::EDGetTokenT<edm::ValueMap<reco::CandidatePtr>>()),
-      PuppiCands_(usePuppi_ ? consumes<std::vector<reco::PFCandidate>>(iConfig.getParameter<edm::InputTag>("PuppiSrc"))
-                            : edm::EDGetTokenT<std::vector<reco::PFCandidate>>()),
+private:
+  // if PuppiSrc && PuppiNoLepSrc are empty, usePuppi is false
+  // otherwise assumes that if they are set, you wanted to use puppi and will
+  // throw an exception if the puppis are not found
+  const bool usePuppi_;
+
+  const edm::EDGetTokenT<reco::PFCandidateCollection> Cands_;
+  const edm::EDGetTokenT<reco::VertexCollection> PVs_;
+  const edm::EDGetTokenT<edm::Association<reco::VertexCollection>> PVAsso_;
+  const edm::EDGetTokenT<edm::ValueMap<int>> PVAssoQuality_;
+  const edm::EDGetTokenT<reco::VertexCollection> PVOrigs_;
+  const edm::EDGetTokenT<reco::TrackCollection> TKOrigs_;
+  const edm::EDGetTokenT<edm::ValueMap<float>> PuppiWeight_;
+  const edm::EDGetTokenT<edm::ValueMap<float>> PuppiWeightNoLep_;
+  const edm::EDGetTokenT<edm::ValueMap<reco::CandidatePtr>> PuppiCandsMap_;
+  const edm::EDGetTokenT<std::vector<reco::PFCandidate>> PuppiCands_;
+  const edm::EDGetTokenT<std::vector<reco::PFCandidate>> PuppiCandsNoLep_;
+  std::vector<edm::EDGetTokenT<edm::View<reco::Candidate>>> SVWhiteLists_;
+  const bool storeChargedHadronIsolation_;
+  const edm::EDGetTokenT<edm::ValueMap<bool>> ChargedHadronIsolation_;
+
+  const double minPtForChargedHadronProperties_;
+  const double minPtForTrackProperties_;
+  const int covarianceVersion_;
+  const std::vector<int> covariancePackingSchemas_;
+
+  const std::vector<int> pfCandidateTypesForHcalDepth_;
+  const bool storeHcalDepthEndcapOnly_;
+
+  const bool storeTiming_;
+
+  // for debugging
+  float calcDxy(float dx, float dy, float phi) const {
+    return -dx * std::sin(phi) + dy * std::cos(phi);
+  }
+  float calcDz(reco::Candidate::Point p, reco::Candidate::Point v,
+               const reco::Candidate &c) const {
+    return p.Z() - v.Z() -
+           ((p.X() - v.X()) * c.px() + (p.Y() - v.Y()) * c.py()) * c.pz() /
+               (c.pt() * c.pt());
+  }
+};
+} // namespace pat
+
+pat::PATPackedCandidateProducer::PATPackedCandidateProducer(
+    const edm::ParameterSet &iConfig)
+    : usePuppi_(
+          !iConfig.getParameter<edm::InputTag>("PuppiSrc").encode().empty() ||
+          !iConfig.getParameter<edm::InputTag>("PuppiNoLepSrc")
+               .encode()
+               .empty()),
```

Comment:
@davidlange6 @fabiocos 
for this and similar cases is there a way to make clang-format not break up similarly looking lines in a different way?
it seems also unnecessary to have ```.encode().empty() ``` broken up to two pretty short lines. Was it intended in the configuration

Reply:
there are not an infinite number of options, no.  This presumably happens because of the greater number of characters in the second line. 

One can try by hand just to have a line break before .encode() on both lines (but that also may get changed by clang-format)